web/src/settings_exports.ts

```diff
@@ -248,3 +405,44 @@ export function set_up(): void {
         });
     });
 }
+
+function maybe_store_export_consent_data_and_return(export_consent: ExportConsent): boolean {
+    // Handles a race where the client has requested the server for export consents
+    // to populate 'Export permissions' table but hasn't received the response yet,
+    // but received a few updated events which should be applied on top of the received
+    // response to avoid outdated table.
+    // We store the export_consent data received via events to apply them on top of
+    // the received response.
+    if (export_consents === undefined) {
+        queued_export_consents.push(export_consent);
+        return true;
+    }
+    return false;
+}
+
+export function remove_export_consent_data_and_redraw(user_id: number): void {
+    if (!meta.loaded) {
+        return;
+    }
+
+    if (export_consents === undefined) {
+        queued_export_consents.push(user_id);
+        return;
+    }
+
+    export_consents.delete(user_id);
+    redraw_export_consents_list();
```

Comment:
This logic could probably be optimized to not redraw in the event that the settings panel is no longer open, since I expect it gets redrawn every time its opened?

(I think `meta.loaded` gets set to true the first time the panel is opened, even if the user navigates elsewhere after?)